Automattic/jetpack
# Publicize: Don't instantiate the global $publicize class multiple times.

The global `$publicize` object is instantiated in the composer package in this file `projects/packages/publicize/src/class-publicize-ui.php`. The composer package is currently used by the plugins Jetpack and Jetpack Social. I am working on a change to get wpcom to use the composer package as well, however wpcom has it own publicize class which shouldn't live in the package and it has its own way to create the global `$publicize` object. However, wpcom will use the the package's Publicize_UI and Publicize_Base classes. This change makes sure that we don't override wpcom's `publicize` object from within the Publicize_UI class. 





#### Changes proposed in this Pull Request:

* Don't instantiate the publicize object if it's already instantiated. 

#### Jetpack product discussion
Check the linked Phabricator patch on this thread - p1654865454855369-slack-C02JJ910CNL. 

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Make sure you're able to Publicize from the Jetpack plugin. 
* Make sure you're able to Publicize from the Jetpack Social plugin.
